src/main/java/uk/gov/ida/matchingserviceadapter/validators/validationrules/ConditionsShouldNotContainOneTimeUseElement.java

```diff
@@ -0,0 +1,18 @@
+package uk.gov.ida.matchingserviceadapter.validators.validationrules;
+
+import org.opensaml.saml.saml2.core.OneTimeUse;
+import uk.gov.ida.matchingserviceadapter.exceptions.SamlResponseValidationException;
+
+public class ConditionsShouldNotContainOneTimeUseElement extends ValidationRule<OneTimeUse> {
+    public ConditionsShouldNotContainOneTimeUseElement() {
```

Comment:
I think this might be better if the generic type was `<Conditions>` and the Predicate was `(e) -> e.getOneTimeUse == null`.  Same for the ProxyRestrictionElement ...